warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyIterator.java

```diff
@@ -324,8 +332,17 @@ public void setKeys(Set<Key> keys) {
     public TreeSet<String> buildUidsFromKeys(Collection<Key> keys) {
         TreeSet<String> uids = new TreeSet<>();
         for (Key key : keys) {
-            DatawaveKey datawaveKey = new DatawaveKey(key);
-            uids.add(datawaveKey.getUid());
+            // keys are CF=datatype\x00uid
+            byte[] backing = key.getColumnFamilyData().getBackingArray();
+            int index = -1;
+            for (int i = 0; i < backing.length; i++) {
```

Comment:
Although backing.length will work, it would be safer to use key.getColumnFamilyData().getLength().

Reply:
No such method?

Reply:
Got it.